### PR TITLE
Remove organisation-level roles

### LINKED ISSUE #4198: Remove organization-level roles
As well as project-level roles, Job Server can record organization-level roles. In practice, it doesn't because:

* they aren't exposed in the UI
* they aren't audited[^1]
* it's unclear how they would interact with project-level roles
* it's unclear whether there is an IG process associated with them

[^1]: The audit log contains entries for three events: _User added [to project]_, _User removed [from project]_, and _User's roles updated [in project]_.

@CLStables suggests either clarifying why organization-level roles exist, and describing the work that would be required to implement them fully, or removing them.

---

@CLStables created this issue as a draft, following a Slack discussion.[^1] It contained two actions, which related to organization roles and the Leaver Checklist. This issue relates to organization roles. #4197 relates to the Leaver Checklist.

### PR DESCRIPTION
**Pull request by Jongmassey**
Fixes #4198  by removing un-used organisation-level roles

### CODE REVIEW

**Review by iaindillingham — COMMENTED** [review]
Looking good! Thanks for your hard work. I've spotted a bug, which needs to be fixed. I'm going to hold-off approving, to prevent accidental merge.

File: tests/unit/jobserver/models/test_user.py [hunk 1]
```diff
@@ -27,7 +26,6 @@ def test_user_all_roles(project_membership):
     project = ProjectFactory(orgs=[org])
     user = UserFactory(roles=[OutputChecker])
 
-    OrgMembershipFactory(org=org, user=user, roles=[OutputChecker])
```
[the remark this answers is not in the document]
Comment (Jongmassey) [reply tests/unit/jobserver/models/test_user.py:30]: Thanks for this, it's almost ironic that my changes to remove testing of org membership roles also removed the bit of the test that would have caught this error! I'll give the sequencing of changes a bit more thought in future.

I've amended the relevant commit to remove org membership roles from `User.all_roles()` and I've double-checked all instances of `OrgMembershipFactory`, `OrgMembership`, and the `org_memberships` relation such that I'm confident there are no further traces of organisation roles.

[the remark this answers is not in the document]
Comment (iaindillingham) [reply tests/unit/jobserver/models/test_user.py:30]: That's great, thank you!

**Comment by Jongmassey** [comment]
> I'm unsure why the checks haven't completed. 


_Helpfully_ the checks don't run if there is a merge conflict with `main`. I've now resolved that conflict so they're running.

**Pull request merged**